discrete: fix DISC_OP_AMP_OSCILLATOR_VCO_3

tylrshrmp opened the pull request:
Hello there,  
I play the Space Invaders on mame, but these sound .... bit differ.  
  
Perhaps this fix the part of the Space Invaders' "TRIGGER SOUND", a beam sound.  
  
This is my first Pull requests. I would be glad to point out any problems.

rb6502 commented on src/devices/sound/disc_wav.hxx:
```diff
@@ -746,7 +746,7 @@ DISCRETE_STEP(dss_op_amp_osc)
 			/* we need to mix any bias and all modulation voltages together. */
 			v = DSS_OP_AMP_OSC__VMOD1 - OP_AMP_NORTON_VBE;
-			if (v < 0) v = 0;
+			// if (v < 0) v = 0; /* no clipping. this voltage can be minus, for charging +/- to "C" */
```
Go ahead and just remove the original if statement here.  The comment you added is fine, but it should be C++ // style.  The explanation sounds reasonable to me with my limited knowledge of circuits, but @couriersud if you're around and could comment that would be great.

tylrshrmp replied:
Thanks for your point out! fixed it.
If approvable, please could you re-open the PR? or should I create a new PR?

tylrshrmp replied:
Opened PR #11287 
If you don't mind, could you please review it?

cuavas commented:
I’m still not sure this change is actually correct  Is the intention for `DISC_OP_AMP_OSCILLATOR_VCO_3` to model circuits with opamps that use single-ended supplies?

It’s used for several Midway sound boards besides Space Invaders, and the source has the following note: “DISC_OP_AMP_OSCILLATOR_VCO_3 doesn't fully represent the actual circuit on a Taito PCB, missile sound is off”.

I think this is going to break the cases where this is the correct circuit model while still not fixing Space Invaders, and just making it sound subjectively “better”.

cuavas commented:
This is the wrong approach.  The model is supposed to simulate a VCO using an opamp with a single-ended supply.  Space Invaders sounds wrong because it’s using this in the place of a VCO circuit that the discrete sound system lacks a model for.  It causes incorrect behaviour for systems where the model is correct.

At this point, the whole discrete sound system is deprecated.  It requires too much interpretation of the circuitry.  It’s too inflexible when it comes to dealing with the variety of circuitry used in real life.

tylrshrmp commented:
Sorry for my late reply. re-opened.

> Is the intention for DISC_OP_AMP_OSCILLATOR_VCO_3 to model circuits with opamps that use single-ended supplies?

Yes, this PR is intended to fix little bugs for electronic circuit simulation statement of it.

`DISC_OP_AMP_OSCILLATOR_VCO_3` is indeed "doesn't fully represent the actual circuit on a Taito PCB", but it was fixed (by this PR) and correctly simulated the "one of single power supply op-amp oscillator circuits, using the Norton-type opamps" as my understand.

I think that that "the actual circuit on a Taito PCB" means
https://www.robotron-2084.co.uk/manuals/invaders/taito_space_invader_l_shaped_board_schematics.pdf
bottom left of page2 in this PDF.
But, in my understand, what `DISC_OP_AMP_OSCILLATOR_VCO_3` is responsible for already fully represented and implemented.
([NOISE -> 0.1uF caps. -> ... LM3900 Norton-opamp marked as "30" and pin-4 output] were implemented by `DISC_OP_AMP_OSCILLATOR_VCO_3`. note to under row of these is the `DISCRETE_OP_AMP_TRIG_VCA` trigger circuit.)

The part of not implemented is the two of inverting amplifier circuits that can disabling by the trigger that after pin-4 outputs. It is certainly not perfect to omit it, but it is an inversion of an inversion, and the part of disabling trigger is maybe no effect (in my understand, point out please if another functions), so it doesn't make much sense and to be a waste of computational cost.

And, the "v" variable has been used for temporary variable that calcurating a "voltage that make current for (dis)charging caps.".
This is a "relative" voltage, so it can be minuses. It is not related to power supply is single-ended.

Therefore,
> This is the wrong approach. The model is supposed to simulate a VCO using an opamp with a single-ended supply. Space Invaders sounds wrong because it’s using this in the place of a VCO circuit that the discrete sound system lacks a model for. It causes incorrect behaviour for systems where the model is correct.

and
> I think this is going to break the cases where this is the correct circuit model while still not fixing Space Invaders, and just making it sound subjectively “better”.

I think it's not. (and I felt a little deserted...)
The changes are few lines, but I feel that I have read and understood the source code and circuits well, fixed the problems, and actually improved it.

But,
> the whole discrete sound system is deprecated. It requires too much interpretation of the circuitry. It’s too inflexible when it comes to dealing with the variety of circuitry used in real life.

I agree very much with this. I want to planning rewrite and replace from the discrete sound to another (netlist-ish) good solution... if I have more time..
(Using the netlist libs is simple and good idea, but it causes more heavilly computational cost... in my thought.)

Sorry for the long post. Best regards.
